trainer.py

```diff
@@ -182,7 +221,29 @@ def compute_vgg_loss(self, vgg, img, target):
         img_fea = vgg(img_vgg)
         target_fea = vgg(target_vgg)
         return torch.mean((self.instancenorm(img_fea) - self.instancenorm(target_fea)) ** 2)
-
+    
+    def compute_semantic_seg_loss(self, img1,img2, mask):
+        # denorm 
+        img1_denorm = (img1 + 1) / 2.
+        img2_denorm = (img2 + 1) / 2.
+        # norm for semantic seg network
+        input_transformed1 = seg_transform()(img1_denorm).unsqueeze(0)
+        input_transformed2 = seg_transform()(img2_denorm).unsqueeze(0)
+        #compute labels from original image and logits from translated version
+        target = self.segmentation_model(input_transformed1).squeeze().max(0)[1].unsqueeze(0)
+        output = self.segmentation_model(input_transformed2)
+        #Resize mask to the size of the image
+        mask1 = torch.nn.functional.interpolate(mask, size=(self.newsize,self.newsize))
+        mask1_tensor = torch.tensor(mask1,dtype=torch.long).cuda()
+        # we want the masked region to be labeled as unknown (19 is not an existing label)
+        target_with_mask = torch.mul(1-mask1_tensor,target) +mask1_tensor*19
+        mask2 = torch.nn.functional.interpolate(mask, size=(self.newsize,self.newsize))
+        mask_tensor = torch.tensor(mask2,dtype=torch.float).cuda()
+        output_with_mask = (torch.mul(1-mask_tensor,output))  
+        # cat the mask as to the logits (loss=0 over the masked region)
+        output_with_mask_cat = torch.cat((output_with_mask.squeeze(),mask_tensor.squeeze().unsqueeze(0)))
+        loss = nn.CrossEntropyLoss()(output_with_mask_cat.unsqueeze(0),target_with_mask.squeeze().unsqueeze(0))
+        return(loss)
     def sample(self, x_a, x_b):
         self.eval()
```

Comment:
comment the function: loading the images for inferences + the segmentation mask before and after translation